qiskit/providers/backend_compat.py

```diff
@@ -158,15 +158,22 @@ def _get_value(prop_dict, prop_name):
         faulty_qubits = {
             q for q in range(configuration.num_qubits) if not properties.is_qubit_operational(q)
         }
-        qubit_properties = [
-            QubitProperties(
-                t1=properties.qubit_property(qubit_idx)["T1"][0],
-                t2=properties.qubit_property(qubit_idx)["T2"][0],
-                frequency=properties.qubit_property(qubit_idx)["frequency"][0],
-            )
-            for qubit_idx in range(0, configuration.num_qubits)
-        ]
 
+        qubit_properties = []
+        for qi in range(0, configuration.num_qubits):
+            # TODO faulty qubit handling might be needed since
+            #  faulty qubit reporting qubit properties doesn't make sense.
```

Comment:
FWIW, looking at the target code: https://github.com/Qiskit/qiskit/blob/945fe1dc508dcedd6cd96ac28e26d524826c748d/qiskit/transpiler/target.py#L318-L326 so not filtering faulty qubits is the expected behavior here. If the number of qubits in the properties dict differs from the number of qubits on the backend it'll error.